iscsi/initiator.go

```diff
@@ -313,43 +310,50 @@ func findScsiDevice(ip, target string, lun int, ne *util.NamespaceExecutor) (*ut
 		return nil, fmt.Errorf("cannot find kernel device for iSCSI device: %s", name)
 	}
 
-	return dev, nil
+	return &dev, nil
 }
 
-func CleanupScsiNodes(target string, ne *util.NamespaceExecutor) error {
+func CleanupScsiNodes(target string) error {
 	for _, dir := range ScsiNodesDirs {
-		if _, err := ne.Execute("ls", []string{dir}); err != nil {
+		if _, err := lhns.GetFileInfo(lhtypes.HostProcDirectory, dir); err != nil {
```

Comment:
Does this mean each lhns will create a new executor for `HostProcDirectory` every time? Can we just use one executor instead? nsenter still a system call, so we should prevent any potential resource usage.

Reply:
`lhns.GetFileInfo` isn't relying on `nsenter` for namespace switching. We've replaced the execution of GNU core utils commands with function calls and used setns for namespace switching.